api/api.proto

```diff
@@ -0,0 +1,73 @@
+syntax = "proto3";
+
+package api;
+
+option go_package = "github.com/ligato/vpp-agent/api";
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+option (gogoproto.messagename_all) = true;
+
+import "google/protobuf/any.proto";
+
+message ModelSpec {
+    string module = 1;
+    string class = 2;
+    string version = 3;
+    string kind = 4;
+}
+
+message Model {
+    google.protobuf.Any any = 1;
+    map<string,string> metadata = 2;
+}
+
+service SyncService {
+    rpc Sync (SyncRequest) returns (SyncResponse) {};
+    rpc Obtain (ObtainRequest) returns (ObtainResponse) {};
+}
+
+message SyncOptions {
+    bool resync = 1;
+    uint32 retry_count = 2;
+}
+
+message SyncRequest {
+    repeated SyncItem items = 1;
+    SyncOptions options = 2;
+}
+
+message SyncResponse {
+    repeated SyncItemResult results = 1;
+}
+
+message SyncItem {
+    Model model = 1;
+    bool delete = 2;
+}
+
+message SyncItemResult {
+    string id = 1;
+    string status = 2;
+    string error = 3;
```

Comment:
Why not simply rely on the normal grpc error mechanism?  Why a separate string here?

Reply:
This is meant to define status/error for the specific model item that was synced. The normal grpc error mechanism will be used for the overall result of the request.

Reply:
Ah... that makes sense